File: Study2021/Programmers_Level_1/로또의 최고 순위와 최저 순위.java
package algorithm;

import java.util.HashSet;
import java.util.Set;

public class Prog77484 {
    public static void main(String[] args) {
        Prog77484 s = new Prog77484();
        int[] lottos = {44, 1, 0, 0, 31, 25};
        int[] win_nums = {31, 10, 45, 1, 6, 19};
        int[] answer = s.solution(lottos, win_nums);
        System.out.printf("%d %d", answer[0], answer[1]);
    }
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int[2];
        int zero_cnt = 0;
        int correct_cnt = 0;

        Set win_nums_set = new HashSet();
        for (int win_num: win_nums) {
            win_nums_set.add(win_num);
        }

        for (int lotto: lottos) {
            if (lotto == 0) {
                zero_cnt++;
            } else if (win_nums_set.contains((lotto))) {
                correct_cnt++;
            }
        }

        answer[0] = getGrade(zero_cnt + correct_cnt);
        answer[1] = getGrade(correct_cnt);
        return answer;
    }

    int getGrade(int cnt) {
        if (cnt > 1) {
            return 7-cnt;
        } else {
            return 6;
        }
    }
}

File: Study2021/Programmers_Level_1/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    answer = []
    zero_cnt = 0
    correct_cnt = 0
    win_nums_set = set(win_nums)

    for lotto in lottos:
        if lotto == 0:
            zero_cnt += 1
        elif lotto in win_nums_set:
            correct_cnt += 1
            
    answer = [get_grade(correct_cnt + zero_cnt), get_grade(correct_cnt)]        
    return answer

def get_grade(cnt):
    if cnt == 6:
        return 1
    elif cnt == 5:
        return 2
    elif cnt == 4:
        return 3
    elif cnt == 3:
        return 4
    elif cnt == 2:
        return 5
    else:
        return 6